src/util/amount.rs

```diff
@@ -68,15 +76,15 @@ impl FromStr for Denomination {
 
     /// Convert from a str to Denomination.
     ///
-    /// Any combination of upper and/or lower case, excluding uppercase 'M' is considered valid.
-    /// - Singular: BTC, mBTC, uBTC
+    /// Any combination of upper and/or lower case, excluding uppercase 'M', 'P' is considered valid.
+    /// - Singular: BTC, mBTC, uBTC, nBTC, pBTC
     /// - Plural or singular: sat, satoshi, bit, msat
     ///
     /// Due to ambiguity between mega and milli we prohibit usage of leading capital 'M'.
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         use self::ParseAmountError::*;
 
-        if s.starts_with('M') {
+        if s.starts_with(|ch| ch == 'M' || ch == 'P') {
```

Comment:
This is correct, you just need to modify the error message handling of `PossiblyConfusingDenomination`

```rust
let (upper, lower) = match d.chars().next() {
    Some('M') => ("Mega", "milli"),
    Some('P') => ("Peta", "pico"),
    // This panic could be avoided by adding enum ConfusingDenomination { Mega, Peta } but is it worth it?
    _ => panic!("invalid error information"),
};
write!(f, "the '{}' at the beginning of {} should technically mean '{}' but that denomination is uncommon and maybe '{}' was intended", d.chars().next().unwrap(), d, upper, lower)
```